File: client/client-core/src/main/java/com/simple/castle/client/game/GameScreen.java
package com.simple.castle.client.game;

import com.simple.castle.core.screen.BaseScreen;

public class GameScreen extends BaseScreen {
//    private final BaseRenderer baseRenderer;
//    private final BaseCamera baseCamera;
//    private final BaseEnvironment baseEnvironment;
//
//    private final ClientSceneManager clientSceneManager;
//
//    private final ServerApi serverApi;
//    private final Model model;
//
//    public GameScreen(BaseRenderer baseRenderer) {
//        this.model = new AssetLoader().loadModel();
//
//        this.baseRenderer = baseRenderer;
//        this.baseCamera = new BaseCamera(new Vector3(10, 10, 10), null);
//        this.baseEnvironment = new BaseEnvironment();
//
//        this.clientSceneManager = new ClientSceneManager(model);
//
//        this.serverApi = new ServerApi(Gdx.net.newClientSocket(Net.Protocol.TCP, "127.0.0.1", 9090, new SocketHints()));
//
//        this.inputMultiplexer.addProcessor(baseCamera);
//    }
//
//    @Override
//    public void render(float delta) {
//        baseCamera.update(delta);
//
//        ServerState tick = serverApi.getPositions();
//
//        Gdx.app.log("GameScreen", tick.toString());
//
//        baseRenderer.render(baseCamera, clientSceneManager.updateAndGet(tick), baseEnvironment);
//    }
//
//    @Override
//    public void resize(int width, int height) {
//        baseCamera.resize(width, height);
//        baseCamera.update();
//    }
//
//    @Override
//    public void dispose() {
//        serverApi.dispose();
//        model.dispose();
//        baseRenderer.dispose();
//    }

}

File: core/common/src/main/kotlin/castle/core/screen/GameScreen.kt
package castle.core.screen

import castle.core.config.GameConfig
import castle.core.screen.add.ScreenConfigurator
import com.badlogic.gdx.Gdx
import ktx.app.KtxScreen

class GameScreen : KtxScreen {
    private val gameConfig = GameConfig()
    private val screenConfigurator = ScreenConfigurator(gameConfig.systems)
    private val engine = screenConfigurator.engine
    private val inputMultiplexer = screenConfigurator.inputMultiplexer
    private val screens = screenConfigurator.screens
    private val disposables = screenConfigurator.disposables

    override fun resize(width: Int, height: Int) {
        screens.forEach { it.resize(width, height) }
    }

    override fun render(delta: Float) {
        engine.update(delta)
    }

    override fun show() {
        Gdx.input.inputProcessor = inputMultiplexer
    }

    override fun hide() {
        Gdx.input.inputProcessor = null
    }

    override fun dispose() {
        engine.removeAllEntities()
        disposables.forEach { it.dispose() }
        gameConfig.dispose()
    }
}